cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1739,37 +1707,17 @@ private boolean canHaveInstances(ProgramType programType) {
   }
 
   // deletes the process metrics for a flow
-  private void deleteProcessMetricsForFlow(String application, String flow) throws IOException {
-    ServiceDiscovered discovered = discoveryServiceClient.discover(Constants.Service.METRICS);
-    Discoverable discoverable = new RandomEndpointStrategy(discovered).pick(3L, TimeUnit.SECONDS);
-
-    if (discoverable == null) {
-      LOG.error("Fail to get any metrics endpoint for deleting metrics.");
-      throw new IOException("Can't find Metrics endpoint");
-    }
-
-    LOG.debug("Deleting metrics for flow {}.{}", application, flow);
-    // TODO: use MetricStore directly to delete the metrics [CDAP-2163]
-    String url = String.format("http://%s:%d%s/metrics/system/apps/%s/flows/%s?prefixEntity=process",
-                               discoverable.getSocketAddress().getHostName(),
-                               discoverable.getSocketAddress().getPort(),
-                               Constants.Gateway.API_VERSION_3,
-                               application, flow);
-
-    long timeout = TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
-
-    SimpleAsyncHttpClient client = new SimpleAsyncHttpClient.Builder()
-      .setUrl(url)
-      .setRequestTimeoutInMs((int) timeout)
-      .build();
-
-    try {
-      client.delete().get(timeout, TimeUnit.MILLISECONDS);
-    } catch (Exception e) {
-      LOG.error("exception making metrics delete call", e);
-      Throwables.propagate(e);
-    } finally {
-      client.close();
-    }
+  private void deleteProcessMetricsForFlow(String namespaceId,
+                                           String application,
+                                           String flow) throws Exception {
+
+    long endTs = System.currentTimeMillis() / 1000;
+    String metricNamePrefix = "process";
+    Map<String, String> tags = Maps.newHashMap();
+    tags.put(Constants.Metrics.Tag.NAMESPACE, namespaceId);
+    tags.put(Constants.Metrics.Tag.APP, application);
+    tags.put(Constants.Metrics.Tag.FLOW, flow);
+    MetricDeleteQuery deleteQuery = new MetricDeleteQuery(0, endTs, metricNamePrefix, tags);
```

Comment:
Current API supports deleting all metrics or specified metric (or specified multiple metrics - after PR #2080 is merged). So, yes, no delete by prefix name.

Not sure if we need to delete queue metrics in this case either. Maybe it is not needed - hence, since it was not doing anything before your change (as delete by prefix is incorrect) - was not spotted as issue..